test/javascripts/acceptance/search-test.js.es6

```diff
@@ -73,3 +73,49 @@ QUnit.test("Search with context", assert => {
     assert.ok(!$('.search-context input[type=checkbox]').is(":checked"));
   });
 });
+
+QUnit.test("Right filters are shown to anonymous users", assert => {
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.ok(exists('select#in option[value=first]'));
+    assert.ok(exists('select#in option[value=pinned]'));
+    assert.ok(exists('select#in option[value=unpinned]'));
+    assert.ok(exists('select#in option[value=wiki]'));
+    assert.ok(exists('select#in option[value=images]'));
```

Comment:
Took the liberty to include the filters from [`inOptionsForAll`](https://github.com/discourse/discourse/pull/5032/files#diff-d8f8a0225b6661e2dd68852bdeb0b9e5R40); making sure they are rendered at all times.